readalongs/web_api.py

```diff
@@ -322,60 +326,93 @@ async def convert_alignment(
     except ValueError as e:
         raise HTTPException(status_code=422, detail="SMIL provided is not valid") from e
 
-    # Data privacy consideration: creating the temporary directory with a context
-    # manager like this guarantees, as we promise in the API documentation, that the
-    # temporary directory and all temporary files we create in it will be deleted
-    # when this function exits, whether it is with an error or with success.
-    with TemporaryDirectory() as temp_dir_name:
-        prefix = os.path.join(temp_dir_name, "aligned")
+    # Data privacy consideration: we have to make sure this temporary directory gets
+    # deleted after the call returns, as we promise in the API documentation.
+    if True:
+        temp_dir_object = tempfile.TemporaryDirectory()
+        temp_dir_name = temp_dir_object.name
+        cleanup = BackgroundTask(temp_dir_object.cleanup)
+    else:
+        temp_dir_name = tempfile.mkdtemp()
+        cleanup = BackgroundTask(lambda: None)
+    prefix = os.path.join(temp_dir_name, "aligned")
+    LOGGER.info(f"Temporary directory: {temp_dir_name}")
 
+    try:
         if output_format == FormatName.textgrid:
-            save_label_files(
-                words, parsed_xml, input.audio_duration, prefix, "textgrid"
-            )
-            return Response(
-                slurp_file(prefix + ".TextGrid"),
+            try:
+                save_label_files(
+                    words, parsed_xml, input.audio_duration, prefix, "textgrid"
+                )
+            except Exception as e:
+                raise HTTPException(
+                    status_code=422,
+                    detail="XML+SMIL file pair provided cannot be converted",
+                ) from e
+            return FileResponse(
+                prefix + ".TextGrid",
+                background=cleanup,
                 media_type="text/plain",
-                # With "attachment;", the result has to be downloaded
-                # headers={"Content-Disposition": "attachment; filename=aligned.TextGrid"},
-                # Without "attachment;", the result is right in the response body
                 headers={"Content-Disposition": "filename=aligned.TextGrid"},
```

Comment:
I think setting `filename="aligned.TextGrid"` is preferred and the `Content-Disposition` header will be set automatically. but this is a very minor point.